### PR TITLE
ext_proc: observability mode part 4: add deferred closure timeout config

### PR DESCRIPTION
**Pull request by tyxia**
The default timeout value is chosen same as [drain timeout](https://github.com/envoyproxy/envoy/blob/3feff04eeb2356cf16f799bc9fb812e5b765315f/api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto#L553) `<envoy_v3_api_field_extensions.filters.network.http_connection_manager.v3.HttpConnectionManager.drain_timeout>`

### CODE REVIEW

**Comment by tyxia** [comment]
/assign @yanavlasov 

Per our discussion.

File: api/envoy/extensions/filters/http/ext_proc/v3/ext_proc.proto [hunk 1]
```diff
@@ -256,6 +256,12 @@ message ExternalProcessor {
   // Only one of ``disable_clear_route_cache`` or ``route_cache_action`` can be set.
   RouteCacheAction route_cache_action = 18
       [(udpa.annotations.field_migrate).oneof_promotion = "clear_route_cache_type"];
+
+  // [#not-implemented-hide:]
+  // Specifies the deferred closure timeout for gRPC stream (that connects to external processor). Currently, deferred stream closure is
+  // only used in :ref:`clear_route_cache <envoy_v3_api_field_extensions.filters.http.ext_proc.v3.ExternalProcessor.observability_mode>`.
```
Comment (htuch) [inline api/envoy/extensions/filters/http/ext_proc/v3/ext_proc.proto:262]:
```suggestion
  // only used in :ref:`observability_mode <envoy_v3_api_field_extensions.filters.http.ext_proc.v3.ExternalProcessor.observability_mode>`.
```

File: api/envoy/extensions/filters/http/ext_proc/v3/ext_proc.proto [hunk 2]
```diff
@@ -256,6 +256,12 @@ message ExternalProcessor {
   // Only one of ``disable_clear_route_cache`` or ``route_cache_action`` can be set.
   RouteCacheAction route_cache_action = 18
       [(udpa.annotations.field_migrate).oneof_promotion = "clear_route_cache_type"];
+
+  // [#not-implemented-hide:]
+  // Specifies the deferred closure timeout for gRPC stream (that connects to external processor). Currently, deferred stream closure is
```
Comment (htuch) [inline api/envoy/extensions/filters/http/ext_proc/v3/ext_proc.proto:261]: I think you should explain in a bit more detail what deferred close is. It needs to be understandable and actionable for a reader who doesn't have deep knowledge of ext_proc filter implementation details, so you need to say something along the lines of "When ext_proc is used in observability mode, streams may be held open to the external processor longer than the lifetime of the regular client to backend stream lifetime. In this case, Envoy will eventually timeout the external processor stream according to this limit". Or something similar along those lines.

Comment (tyxia) [reply api/envoy/extensions/filters/http/ext_proc/v3/ext_proc.proto]: thank you for suggested comments !

Done.

File: api/envoy/extensions/filters/http/ext_proc/v3/ext_proc.proto [hunk 3]
```diff
@@ -256,6 +256,14 @@ message ExternalProcessor {
   // Only one of ``disable_clear_route_cache`` or ``route_cache_action`` can be set.
   RouteCacheAction route_cache_action = 18
       [(udpa.annotations.field_migrate).oneof_promotion = "clear_route_cache_type"];
+
+  // [#not-implemented-hide:]
+  // Specifies the deferred closure timeout for gRPC stream that connects to external processor. Currently, the deferred stream closure
+  // is only used in :ref:`observability_mode <envoy_v3_api_field_extensions.filters.http.ext_proc.v3.ExternalProcessor.observability_mode>`.
+  // In observability mode, grpc streams may be held open to the external processor longer than the lifetime of the regular client to
```
Comment (htuch) [inline api/envoy/extensions/filters/http/ext_proc/v3/ext_proc.proto:263]: Nit: gRPC

**Review by htuch — APPROVED** [review]
LGTM, will merge unless @mattklein123 has any objections.

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**